src/sentry/api/endpoints/organization_member/index.py

```diff
@@ -371,27 +375,39 @@ def post(self, request: Request, organization) -> Response:
                 status=400,
             )
 
-        with transaction.atomic(router.db_for_write(OrganizationMember)):
-            # remove any invitation requests for this email before inviting
-            existing_invite = OrganizationMember.objects.filter(
-                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
-                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
-                email=result["email"],
-                organization=organization,
-            )
-            for om in existing_invite:
-                om.delete()
-
-            om = OrganizationMember(
-                organization=organization,
-                email=result["email"],
-                role=result["role"],
-                inviter_id=request.user.id,
-            )
+        if is_active_staff(request):
+            with transaction.atomic(router.db_for_write(OrganizationMember)):
```

Comment:
nit: you could pull the transaction out to the top layer since you need it for both if/else